server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -101,42 +120,23 @@ public boolean isRolledOver() {
         return rolledOver;
     }
 
-    /**
-     * Returns true if the creation of the new rollover index and switching of the
-     * alias to the newly created index was successful, and returns false otherwise.
-     * If {@link #isDryRun()} is true, then this will also return false. If this
-     * returns false, then {@link #isShardsAcknowledged()} will also return false.
-     */
-    public boolean isAcknowledged() {
-        return acknowledged;
-    }
-
-    /**
-     * Returns true if the requisite number of shards were started in the newly
-     * created rollover index before returning. If {@link #isAcknowledged()} is
-     * false, then this will also return false.
-     */
-    public boolean isShardsAcknowledged() {
-        return shardsAcknowledged;
-    }
-
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         oldIndex = in.readString();
         newIndex = in.readString();
         int conditionSize = in.readVInt();
-        Set<Map.Entry<String, Boolean>> conditions = new HashSet<>(conditionSize);
+        Map<String, Boolean> conditions = new HashMap<>(conditionSize);
         for (int i = 0; i < conditionSize; i++) {
             String condition = in.readString();
             boolean satisfied = in.readBoolean();
-            conditions.add(new AbstractMap.SimpleEntry<>(condition, satisfied));
+            conditions.put(condition, satisfied);
         }
         conditionStatus = conditions;
```

Comment:
nit: wonder why we need the intermediate "condition" map here.

Reply:
I don't follow, you mean why we need the two intermediate variables?

Reply:
I mean the "conditions" map, which is created locally here and then assigned to conditionStatus. We could initialize conditionStatus directly and read into it instead, unless this is done for some concurrency reason (which in this case I doubt).

Reply:
oh good one, I didn't see it before :) thanks